formats/utils/role-variants.js

```diff
@@ -0,0 +1,745 @@
+const hueRotationFn = (rotation) => {
+  return (hue) => (360 + hue + rotation) % 360;
+};
+
+const saturationAdjustmentFn = (adjustment) => {
+  return (saturation) => Math.min(Math.max(saturation + adjustment, 0), 100);
+};
+
+const roleVariants = {
```

Comment:
Wondering if we should name this file something like `back-office-variants` or something like that?